app/lib/task/backend.dart

```diff
@@ -326,6 +332,11 @@ class TaskBackend {
       // be relevant to the next iteration.
       seen.removeWhere((_, updated) => updated.isBefore(since));
 
+      // Do not start waiting if after that we would be over the stop threshold.
+      if (clock.fromNow(minutes: 10).isAfter(stopAfter)) {
+        break;
+      }
```

Comment:
Yes, it'll cause it to restart the scanning, but it'll also release the lock, so someone will have to grab the lock again. It'll probably be the same process, but it's just seems like releasing the lock is unnecessary. Maybe I'm overthinking it.

```dart
var nextLongScan = clock.fromNow(hours: 6);

...
var since = clock.ago(days: 3);
while (claim.valid && !abort.isCompleted) {
  ...

  if (clock.now().isAfter(nextLongScan)) {
    // Next time we'll do a longer scan
    since = clock.ago(days: 1);
    nextLongScan = clock.fromNow(hours: 6);
  } else {
    // Next time we'll only consider changes since now - 5 minutes
    since = clock.ago(minutes: 5);
  }

  ...
```

It probably does about the same..

Reply:
Updated.